### PR TITLE
vendor: use github.com/blang/semver/v4

### PR DESCRIPTION
**Pull request by tklauser**
Use the v4 API of `github.com/blang/semver` which is implemented as a
properly versioned Go module.

While at it also get rid of renamed imports.

In the dev-doctor, this also allows to drop the regexp used to strip
leading zeroes from the version number before passing it to
`semver.ParseTolerant`, since the v4 func now does that as well.

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

**Review by rolinh — APPROVED** [review]

**Review by jibi — APPROVED** [review]

**Review by aanm — APPROVED** [review]
:rocket:

File: tools/dev-doctor/binarycheck.go [hunk 1]
```diff
@@ -79,15 +75,9 @@ func (c *binaryCheck) Run() (checkResult, string) {
...
+		// github.com/blang/semver/v4 will understand
 		versionBytes := origVersion
 
-		// sadly github.com/blang/semver.ParseTolerant is not tolerant of
-		// leading zeros in version numbers, as used by Docker, e.g. "19.03.13".
-		// so, remove leading zeros with a regexp. sorry, this is horrible but
-		// the alternative was bringing in a new version number handling module
-		versionBytes = leadingZerosRegexp.ReplaceAll(versionBytes, []byte("$1"))
-
 		version, err := semver.ParseTolerant(string(versionBytes))
```
Comment (twpayne) [inline tools/dev-doctor/binarycheck.go]: `versionBytes` can be dropped completely here - it's the same value as `origVersion` at this point. `origVersion` can then be renamed to just `version`.

**Comment by tklauser** [comment]
test-me-please

**Review by twpayne — APPROVED** [review]
LGTM

**Comment by tklauser** [comment]
retest-runtime

**Review by fristonio — APPROVED** [review]
LGTM :rocket:

**Comment by pchaigno** [comment]
We have a good number of reviews and only @cilium/ci-structure is missing a review. Only one modified file belongs to that team and it received minor changes. The Runtime tests are also failing because of known flake fixed in https://github.com/cilium/cilium/pull/14335. Marking as ready to merge.

**Review by nebril — APPROVED** [review]

**Pull request merged**